app/controllers/student_controller.rb

```diff
@@ -1,5 +1,10 @@
 class StudentController < ApplicationController
 
   def lessons
+    if current_student
+      @student_id = current_student.id
+    else
+      redirect_to new_student_session_path
```

Comment:
instead of this else, you can just add ``` before_action :authenticate_student!
``` at the top of the class

check admincontroller for reference